packages/web-react/src/components/TextField/__tests__/TextField.test.tsx

```diff
@@ -50,6 +50,34 @@ describe('TextField', () => {
       expect(element.textContent).toBe('text');
     });
 
+    it('should have password toggle', () => {
+      const dom = render(<TextField id="textfield" label="Label" hasPasswordToggle />);
+
+      const element = dom.container.querySelector('.TextField__passwordToggle') as HTMLElement;
+      expect(element).toBeInTheDocument();
+    });
+
+    it('should have password toggle button', () => {
+      const dom = render(<TextField id="textfield" label="Label" hasPasswordToggle />);
+
+      const element = dom.container.querySelector('.TextField__passwordToggle__button') as HTMLElement;
+      expect(element).toBeInTheDocument();
+    });
+
+    it('should have type password with password toggle', () => {
+      const dom = render(<TextField id="textfield" label="Label" hasPasswordToggle />);
+
+      const element = dom.container.querySelector('input') as HTMLElement;
+      expect(element).toHaveAttribute('type', 'password');
+    });
+
+    it('should have correct aria label of the password toggle', () => {
+      const dom = render(<TextField id="textfield" label="Label" hasPasswordToggle />);
+
+      const element = dom.container.querySelector('.TextField__passwordToggle__button') as HTMLElement;
+      expect(element).toHaveAttribute('aria-label', 'Show password');
+    });
+
```

Comment:
Please add `fireEvent` and further extend the tests with the click state when passwordToggle is.
```suggestion
    it('should have type password with password toggle', () => {
      const dom = render(<TextField id="textfield" label="Label" hasPasswordToggle />);

      const element = dom.container.querySelector('input') as HTMLElement;
      const trigger = dom.container.querySelector('.TextField__passwordToggle__button') as HTMLElement;

      expect(element).toHaveAttribute('type', 'password');

      fireEvent.click(trigger);

      expect(element).toHaveAttribute('type', 'text');
    });

    it('should have correct aria label of the password toggle', () => {
      const dom = render(<TextField id="textfield" label="Label" hasPasswordToggle />);

      const element = dom.container.querySelector('.TextField__passwordToggle__button') as HTMLElement;
      expect(element).toHaveAttribute('aria-label', 'Show password');

      fireEvent.click(element);

      expect(element).toHaveAttribute('aria-label', 'Hide password');
    });
```